### ISSUE TITLE
How do I copy a magnet link of a download?

### ISSUE THREAD

**Issue by vi** [Tribler/tribler#4443]
Or how do I populate a channel from downloads?

##### Tribler version/branch+revision: #####

release-7.3.0-beta1, 65d974cfa262d55f342fb06d6ca1c7b7d6f82e0d

##### Operating system and version: #####

Linux

##### Steps to reproduce the behavior: #####

1. Create a channel
2. Inspect "Add" menu. Looks similar to the menu about adding a download. Settle on "Add torrent from URL/magnet link" (not click although).
3. Go to downloads
4. Select a download.
5. Inspect "Details" information  at the bottom.
6. Look at "infohash". Try to paste it as an URL for channel. Obviously, fails.
7. Open context menu on a download.

##### Expected behavior: #####

There is a magnet link to the download in "Details", or there is something like "Copy magnet link" in context menu.

If it is not possible, then there should be a way to add to channel by infohash.

##### Actual behavior: #####

No obvious way to add a download to a channel without fully re-importing the file.

**Comment by xoriole** [comment]
Related https://github.com/Tribler/tribler/issues/4390.
Partly fixed by https://github.com/Tribler/tribler/pull/4442.

**Comment by vi** [comment]
What about getting magnet link suitable for non-Tribler downloads?

**Comment by xoriole** [comment]
PR #4442 adds this copy button in the details tab. So this option to copy will be available everywhere the details tab is present.
![magnet-link](https://user-images.githubusercontent.com/1442867/55146389-c79b9900-5144-11e9-90a1-8d9d7f310d6f.png)

**Comment by vi** [comment]
It looks like that [Copy] button it would the infohash, i.e. "8427..707", not "magnet:...".

If it indeed copies the magnet link, the button should ba called [Copy magnet link].

P.S. How are infohashes and magnet link related? I assumed "infohash" is a Tribler-specific term.

**Comment by qstokkink** [comment]
The info hash is actually formally the _BitTorrent info hash_, which is the `btih` part of magnet links. The info hashes capture the file names, file paths and file contents in torrents.

The magnet URI format allows you to also add some additional information like a torrent name (which the bittorrent protocol would otherwise need to resolve from a DHT lookup). If you want the full story, you can read more about this [on the maget URI Wikipedia page](https://en.wikipedia.org/wiki/Magnet_URI_scheme).

For convenience, we also recently added the feature to add torrents by just the info hash.

**Comment by vi** [comment]
Is this supposed to work?:

1. Select some download in "Seeding" state.
2. Copy infohash.
3. Prepend `magnet:?xt=urn:btih:` to it.
4. Attempt to add a download from that magnet link

Expected:

It shows files, maybe says it is already downloaded.

Actual:

"Loading files..." seemingly forever.

**Comment by qstokkink** [comment]
Yes that works, it will load the file metadata from the DHT. Do note that this can take quite a while (or not complete at all), as you are dependent on the other people in the network actually having this information.

**Comment by vi** [comment]
> dependent on the other people in the network actually having this information

But it's my own seeding download? How can't I find my own information?

**Comment by qstokkink** [comment]
I would have to speculate as to why you can't find yourself, as I don't know how `libtorrent` performs its DHT lookups exactly. What I can say, is that you are always part of the DHT (though indirectly when performing our special Tribler anonymous downloads).

If it is a [NAT puncturing](https://en.wikipedia.org/wiki/UDP_hole_punching) issue, anonymously seeding should help in you discovering your own information.

**Comment by vi** [comment]
I tried that both with non-anonymous and 1-hop download.

By the way, if I proceed adding a download without waiting for file list to load, I acts strange:

* Original download disappears from the list.
* New "Unknown name" download appears in the list, it has:
* 0B size
* "Seeding" state
* non-zero Seeds and Peers counters
* non-zero Ratio
* current "Added on" data.
* no files

Expected: message about duplicate download.

Restarting Tribler showed status "Waiting for peers" for that download, then it was re-checked, then started Seeding again. Ratio reset to zero.
